### ISSUE TITLE
Loading a plugin without `ui/index.js` fails

### ISSUE THREAD

**Issue by mzbroch** [nautobot/nautobot#3829]
### Environment
* Nautobot version (Docker tag too if applicable): **next**
* Python version: 3.8
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Create Nautobot plugin without `ui/index.js` file in it.
2. Attempt to load Nautobot 2.0 (#next) with ☝️ plugin enabled.

### Expected Behavior

Nautobot starts successfully.

### Observed Behavior

Nautobot fails to start:

```
nautobot_1  | >>> Rendering Nautobot App imports...
nautobot_1  | - App 'design_builder' does not publish a UI; Skipping...
nautobot_1  | Traceback (most recent call last):
nautobot_1  |   File "/usr/local/bin/nautobot-server", line 8, in <module>
nautobot_1  |     sys.exit(main())
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/cli/__init__.py", line 54, in main
nautobot_1  |     run_app(
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
nautobot_1  |     management.execute_from_command_line([runner_name, command] + command_args)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
nautobot_1  |     utility.execute()
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
nautobot_1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
nautobot_1  |     self.execute(*args, **cmd_options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
nautobot_1  |     output = self.handle(*args, **options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/post_upgrade.py", line 95, in handle
nautobot_1  |     call_command("build_ui", npm_install=True)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 181, in call_command
nautobot_1  |     return command.execute(*args, **defaults)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
nautobot_1  |     output = self.handle(*args, **options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 230, in handle
nautobot_1  |     self.render_app_imports()
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 148, in render_app_imports
nautobot_1  |     app_routes[app_name] = self.render_routes_imports(abs_app_path, app_name, app_config)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 114, in render_routes_imports
nautobot_1  |     if component := self.get_app_component(
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 76, in get_app_component
nautobot_1  |     with open(file_path, "r") as f:
nautobot_1  | FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.8/site-packages/nautobot_data_validation_engine/ui/index.js'
nautobot_1  | ⏳ Waiting on DB... (9s / 30s)
```

**Comment by bryanculver** [comment]
Closed with #3833

**Comment by mzbroch** [comment]
@bryanculver It still fails with FileNotFoundError exception :

```
nautobot_1  | >>> Rendering Nautobot App imports...
nautobot_1  | - App 'design_builder' does not publish a UI; Skipping...
nautobot_1  | Traceback (most recent call last):
nautobot_1  |   File "/usr/local/bin/nautobot-server", line 8, in <module>
nautobot_1  |     sys.exit(main())
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/cli/__init__.py", line 54, in main
nautobot_1  |     run_app(
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
nautobot_1  |     management.execute_from_command_line([runner_name, command] + command_args)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
nautobot_1  |     utility.execute()
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
nautobot_1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
nautobot_1  |     self.execute(*args, **cmd_options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
nautobot_1  |     output = self.handle(*args, **options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/post_upgrade.py", line 95, in handle
nautobot_1  |     call_command("build_ui", npm_install=True)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 181, in call_command
nautobot_1  |     return command.execute(*args, **defaults)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
nautobot_1  |     output = self.handle(*args, **options)
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 235, in handle
nautobot_1  |     self.render_app_imports()
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 152, in render_app_imports
nautobot_1  |     if app_routes_imports := self.render_routes_imports(abs_app_path, app_name, app_config):
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 115, in render_routes_imports
nautobot_1  |     if component := self.get_app_component(
nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 76, in get_app_component
nautobot_1  |     with open(file_path, "r") as f:
nautobot_1  | FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.8/site-packages/nautobot_data_validation_engine/ui/index.js'
```

**Comment by bryanculver** [comment]
> @bryanculver It still fails with FileNotFoundError exception :
> 
> ```
> nautobot_1  | >>> Rendering Nautobot App imports...
> nautobot_1  | - App 'design_builder' does not publish a UI; Skipping...
> nautobot_1  | Traceback (most recent call last):
> nautobot_1  |   File "/usr/local/bin/nautobot-server", line 8, in <module>
> nautobot_1  |     sys.exit(main())
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/cli/__init__.py", line 54, in main
> nautobot_1  |     run_app(
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/runner/runner.py", line 263, in run_app
> nautobot_1  |     management.execute_from_command_line([runner_name, command] + command_args)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
> nautobot_1  |     utility.execute()
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
> nautobot_1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
> nautobot_1  |     self.execute(*args, **cmd_options)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
> nautobot_1  |     output = self.handle(*args, **options)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/post_upgrade.py", line 95, in handle
> nautobot_1  |     call_command("build_ui", npm_install=True)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 181, in call_command
> nautobot_1  |     return command.execute(*args, **defaults)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
> nautobot_1  |     output = self.handle(*args, **options)
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 235, in handle
> nautobot_1  |     self.render_app_imports()
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 152, in render_app_imports
> nautobot_1  |     if app_routes_imports := self.render_routes_imports(abs_app_path, app_name, app_config):
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 115, in render_routes_imports
> nautobot_1  |     if component := self.get_app_component(
> nautobot_1  |   File "/usr/local/lib/python3.8/site-packages/nautobot/core/management/commands/build_ui.py", line 76, in get_app_component
> nautobot_1  |     with open(file_path, "r") as f:
> nautobot_1  | FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.8/site-packages/nautobot_data_validation_engine/ui/index.js'
> ```

🤦 @timizuoebideri1 Looks like we missed `FileNotFoundError` here: https://github.com/nautobot/nautobot/blob/f704094b77364b3292dbd12f6aa18626071a3e17/nautobot/core/management/commands/build_ui.py#L123

**Comment by bryanculver** [comment]
Closed again with #3853
